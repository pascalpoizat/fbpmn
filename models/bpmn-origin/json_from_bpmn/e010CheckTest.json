{
    "isInterrupt": {},
    "catN": {
        "Task_1eirt50": "AbstractTask",
        "StartEvent_13qwxem": "NoneStartEvent",
        "Task_1rt44mz": "SendTask",
        "StartEvent_1": "NoneStartEvent",
        "B_": "Process",
        "EndEvent_1s2d234": "NoneEndEvent",
        "SP": "SubProcess",
        "EndEvent_0v9lt5i": "NoneEndEvent",
        "ExclusiveGateway_0079typ": "XorGateway",
        "ReceiveTask_0tuz5r7": "ReceiveTask",
        "Task_097548f": "AbstractTask",
        "ExclusiveGateway_1j1chqb": "AndGateway",
        "EndEvent_0ns8te3": "NoneEndEvent",
        "A_": "Process",
        "StartEvent_0ayuyd2": "NoneStartEvent"
    },
    "catE": {
        "SequenceFlow_0ehu3gz": "NormalSequenceFlow",
        "SequenceFlow_1rma85g": "NormalSequenceFlow",
        "mf1": "MessageFlow",
        "SequenceFlow_0z2xwql": "NormalSequenceFlow",
        "SequenceFlow_00aes3w": "NormalSequenceFlow",
        "SequenceFlow_1vue23p": "NormalSequenceFlow",
        "SequenceFlow_0uzla8o": "NormalSequenceFlow",
        "SequenceFlow_0wto9d1": "NormalSequenceFlow",
        "SequenceFlow_0b7efwa": "NormalSequenceFlow",
        "SequenceFlow_0uplc1a": "NormalSequenceFlow",
        "SequenceFlow_1fuwd1z": "NormalSequenceFlow",
        "SequenceFlow_01wc4ks": "NormalSequenceFlow"
    },
    "containN": {
        "B_": [
            "EndEvent_1s2d234",
            "StartEvent_13qwxem",
            "ReceiveTask_0tuz5r7"
        ],
        "SP": [
            "EndEvent_0v9lt5i",
            "Task_1eirt50",
            "Task_097548f",
            "ExclusiveGateway_1j1chqb",
            "StartEvent_1",
            "ExclusiveGateway_0079typ",
            "Task_1rt44mz"
        ],
        "A_": [
            "SP",
            "StartEvent_0ayuyd2",
            "EndEvent_0ns8te3"
        ]
    },
    "messageE": {
        "mf1": "m1"
    },
    "timeInformation": {},
    "targetE": {
        "SequenceFlow_0ehu3gz": "EndEvent_1s2d234",
        "SequenceFlow_1rma85g": "ReceiveTask_0tuz5r7",
        "mf1": "ReceiveTask_0tuz5r7",
        "SequenceFlow_0z2xwql": "ExclusiveGateway_1j1chqb",
        "SequenceFlow_00aes3w": "EndEvent_0ns8te3",
        "SequenceFlow_1vue23p": "SP",
        "SequenceFlow_0uzla8o": "EndEvent_0v9lt5i",
        "SequenceFlow_0wto9d1": "Task_1eirt50",
        "SequenceFlow_0b7efwa": "ExclusiveGateway_0079typ",
        "SequenceFlow_0uplc1a": "Task_097548f",
        "SequenceFlow_1fuwd1z": "Task_1rt44mz",
        "SequenceFlow_01wc4ks": "ExclusiveGateway_0079typ"
    },
    "name": "e010CheckTest",
    "attached": {},
    "containE": {
        "B_": [
            "SequenceFlow_0ehu3gz",
            "SequenceFlow_1rma85g"
        ],
        "SP": [
            "SequenceFlow_0z2xwql",
            "SequenceFlow_0wto9d1",
            "SequenceFlow_0uplc1a",
            "SequenceFlow_01wc4ks",
            "SequenceFlow_0b7efwa",
            "SequenceFlow_1fuwd1z",
            "SequenceFlow_0uzla8o"
        ],
        "A_": [
            "SequenceFlow_1vue23p",
            "SequenceFlow_00aes3w"
        ]
    },
    "edges": [
        "mf1",
        "SequenceFlow_1vue23p",
        "SequenceFlow_00aes3w",
        "SequenceFlow_0z2xwql",
        "SequenceFlow_0wto9d1",
        "SequenceFlow_0uplc1a",
        "SequenceFlow_01wc4ks",
        "SequenceFlow_0b7efwa",
        "SequenceFlow_1fuwd1z",
        "SequenceFlow_0uzla8o",
        "SequenceFlow_0ehu3gz",
        "SequenceFlow_1rma85g"
    ],
    "messages": [
        "m1"
    ],
    "sourceE": {
        "SequenceFlow_0ehu3gz": "ReceiveTask_0tuz5r7",
        "SequenceFlow_1rma85g": "StartEvent_13qwxem",
        "mf1": "Task_1rt44mz",
        "SequenceFlow_0z2xwql": "StartEvent_1",
        "SequenceFlow_00aes3w": "SP",
        "SequenceFlow_1vue23p": "StartEvent_0ayuyd2",
        "SequenceFlow_0uzla8o": "Task_1rt44mz",
        "SequenceFlow_0wto9d1": "ExclusiveGateway_1j1chqb",
        "SequenceFlow_0b7efwa": "Task_1eirt50",
        "SequenceFlow_0uplc1a": "ExclusiveGateway_1j1chqb",
        "SequenceFlow_1fuwd1z": "ExclusiveGateway_0079typ",
        "SequenceFlow_01wc4ks": "Task_097548f"
    },
    "nodes": [
        "A_",
        "B_",
        "SP",
        "StartEvent_0ayuyd2",
        "EndEvent_0ns8te3",
        "EndEvent_0v9lt5i",
        "Task_1eirt50",
        "Task_097548f",
        "ExclusiveGateway_1j1chqb",
        "StartEvent_1",
        "ExclusiveGateway_0079typ",
        "Task_1rt44mz",
        "EndEvent_1s2d234",
        "StartEvent_13qwxem",
        "ReceiveTask_0tuz5r7"
    ],
    "nameN": {
        "Task_1eirt50": "Task A",
        "Task_1rt44mz": "Task C",
        "ReceiveTask_0tuz5r7": "Task D",
        "Task_097548f": "Task B"
    }
}