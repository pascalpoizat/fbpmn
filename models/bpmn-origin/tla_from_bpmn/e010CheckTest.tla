---------------- MODULE e010CheckTest ----------------

EXTENDS TLC, PWSTypes

VARIABLES nodemarks, edgemarks, net

Interest ==
  "A_" :> {  }
  @@ "B_" :> { "m1" }

ContainRel ==
  "A_" :> { "SP", "StartEvent_0ayuyd2", "EndEvent_0ns8te3" }
  @@ "B_" :> { "EndEvent_1s2d234", "StartEvent_13qwxem", "ReceiveTask_0tuz5r7" }
  @@ "SP" :> { "EndEvent_0v9lt5i", "Task_1eirt50", "Task_097548f", "ExclusiveGateway_1j1chqb", "StartEvent_1", "ExclusiveGateway_0079typ", "Task_1rt44mz" }

Node == { "A_", "B_", "SP", "StartEvent_0ayuyd2", "EndEvent_0ns8te3", "EndEvent_0v9lt5i", "Task_1eirt50", "Task_097548f", "ExclusiveGateway_1j1chqb", "StartEvent_1", "ExclusiveGateway_0079typ", "Task_1rt44mz", "EndEvent_1s2d234", "StartEvent_13qwxem", "ReceiveTask_0tuz5r7" }

Edge == { "mf1", "SequenceFlow_1vue23p", "SequenceFlow_00aes3w", "SequenceFlow_0z2xwql", "SequenceFlow_0wto9d1", "SequenceFlow_0uplc1a", "SequenceFlow_01wc4ks", "SequenceFlow_0b7efwa", "SequenceFlow_1fuwd1z", "SequenceFlow_0uzla8o", "SequenceFlow_0ehu3gz", "SequenceFlow_1rma85g" }

Message == { "m1" }

msgtype ==
   "mf1" :> "m1"


source ==
   "mf1" :> "Task_1rt44mz"
@@ "SequenceFlow_1vue23p" :> "StartEvent_0ayuyd2"
@@ "SequenceFlow_00aes3w" :> "SP"
@@ "SequenceFlow_0z2xwql" :> "StartEvent_1"
@@ "SequenceFlow_0wto9d1" :> "ExclusiveGateway_1j1chqb"
@@ "SequenceFlow_0uplc1a" :> "ExclusiveGateway_1j1chqb"
@@ "SequenceFlow_01wc4ks" :> "Task_097548f"
@@ "SequenceFlow_0b7efwa" :> "Task_1eirt50"
@@ "SequenceFlow_1fuwd1z" :> "ExclusiveGateway_0079typ"
@@ "SequenceFlow_0uzla8o" :> "Task_1rt44mz"
@@ "SequenceFlow_0ehu3gz" :> "ReceiveTask_0tuz5r7"
@@ "SequenceFlow_1rma85g" :> "StartEvent_13qwxem"

target ==
   "mf1" :> "ReceiveTask_0tuz5r7"
@@ "SequenceFlow_1vue23p" :> "SP"
@@ "SequenceFlow_00aes3w" :> "EndEvent_0ns8te3"
@@ "SequenceFlow_0z2xwql" :> "ExclusiveGateway_1j1chqb"
@@ "SequenceFlow_0wto9d1" :> "Task_1eirt50"
@@ "SequenceFlow_0uplc1a" :> "Task_097548f"
@@ "SequenceFlow_01wc4ks" :> "ExclusiveGateway_0079typ"
@@ "SequenceFlow_0b7efwa" :> "ExclusiveGateway_0079typ"
@@ "SequenceFlow_1fuwd1z" :> "Task_1rt44mz"
@@ "SequenceFlow_0uzla8o" :> "EndEvent_0v9lt5i"
@@ "SequenceFlow_0ehu3gz" :> "EndEvent_1s2d234"
@@ "SequenceFlow_1rma85g" :> "ReceiveTask_0tuz5r7"

CatN ==
   "A_" :> Process
@@ "B_" :> Process
@@ "SP" :> SubProcess
@@ "StartEvent_0ayuyd2" :> NoneStartEvent
@@ "EndEvent_0ns8te3" :> NoneEndEvent
@@ "EndEvent_0v9lt5i" :> NoneEndEvent
@@ "Task_1eirt50" :> AbstractTask
@@ "Task_097548f" :> AbstractTask
@@ "ExclusiveGateway_1j1chqb" :> Parallel
@@ "StartEvent_1" :> NoneStartEvent
@@ "ExclusiveGateway_0079typ" :> ExclusiveOr
@@ "Task_1rt44mz" :> SendTask
@@ "EndEvent_1s2d234" :> NoneEndEvent
@@ "StartEvent_13qwxem" :> NoneStartEvent
@@ "ReceiveTask_0tuz5r7" :> ReceiveTask

CatE ==
   "mf1" :> MessageFlow
@@ "SequenceFlow_1vue23p" :> NormalSeqFlow
@@ "SequenceFlow_00aes3w" :> NormalSeqFlow
@@ "SequenceFlow_0z2xwql" :> NormalSeqFlow
@@ "SequenceFlow_0wto9d1" :> NormalSeqFlow
@@ "SequenceFlow_0uplc1a" :> NormalSeqFlow
@@ "SequenceFlow_01wc4ks" :> NormalSeqFlow
@@ "SequenceFlow_0b7efwa" :> NormalSeqFlow
@@ "SequenceFlow_1fuwd1z" :> NormalSeqFlow
@@ "SequenceFlow_0uzla8o" :> NormalSeqFlow
@@ "SequenceFlow_0ehu3gz" :> NormalSeqFlow
@@ "SequenceFlow_1rma85g" :> NormalSeqFlow

LOCAL preEdges ==
  [ i \in {} |-> {}]
PreEdges(n,e) == preEdges[n,e]

PreNodes(n,e) == { target[ee] : ee \in preEdges[n,e] }
          \union { nn \in { source[ee] : ee \in preEdges[n,e] } : CatN[nn] \in { NoneStartEvent, MessageStartEvent } }

BoundaryEvent ==
  [ i \in {} |-> {}]

WF == INSTANCE PWSWellFormed
ASSUME WF!WellTyped
ASSUME WF!WellFormedness

ConstraintNode == TRUE \* none
ConstraintEdge == TRUE \* none
Constraint == TRUE     \* none
INSTANCE PWSConstraints
INSTANCE UserProperties
INSTANCE PWSSemantics

================================================================

