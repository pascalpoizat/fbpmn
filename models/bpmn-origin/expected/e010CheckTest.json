{
  "e010CheckTest": {
    "Network02FifoPair": {
      "Prop02Safe": {
        "value": false,
        "states": null,
        "trans": null,
        "depth": "11"
      },
      "Prop04MsgSound": {
        "value": true,
        "states": "186",
        "trans": "423",
        "depth": "23"
      },
      "Prop03Sound": {
        "value": false,
        "states": "186",
        "trans": "423",
        "depth": null
      },
      "Prop01Type": {
        "value": true,
        "states": "186",
        "trans": "423",
        "depth": "23"
      }
    },
    "Network03Causal": {
      "Prop02Safe": {
        "value": false,
        "states": null,
        "trans": null,
        "depth": "11"
      },
      "Prop04MsgSound": {
        "value": true,
        "states": "186",
        "trans": "423",
        "depth": "23"
      },
      "Prop03Sound": {
        "value": false,
        "states": "186",
        "trans": "423",
        "depth": null
      },
      "Prop01Type": {
        "value": true,
        "states": "186",
        "trans": "423",
        "depth": "23"
      }
    },
    "Network07RSC": {
      "Prop02Safe": {
        "value": false,
        "states": null,
        "trans": null,
        "depth": "11"
      },
      "Prop04MsgSound": {
        "value": true,
        "states": "171",
        "trans": "390",
        "depth": "23"
      },
      "Prop03Sound": {
        "value": false,
        "states": "171",
        "trans": "390",
        "depth": null
      },
      "Prop01Type": {
        "value": true,
        "states": "171",
        "trans": "390",
        "depth": "23"
      }
    },
    "Network05Outbox": {
      "Prop02Safe": {
        "value": false,
        "states": null,
        "trans": null,
        "depth": "11"
      },
      "Prop04MsgSound": {
        "value": true,
        "states": "186",
        "trans": "423",
        "depth": "23"
      },
      "Prop03Sound": {
        "value": false,
        "states": "186",
        "trans": "423",
        "depth": null
      },
      "Prop01Type": {
        "value": true,
        "states": "186",
        "trans": "423",
        "depth": "23"
      }
    },
    "Network04Inbox": {
      "Prop02Safe": {
        "value": false,
        "states": null,
        "trans": null,
        "depth": "11"
      },
      "Prop04MsgSound": {
        "value": true,
        "states": "186",
        "trans": "423",
        "depth": "23"
      },
      "Prop03Sound": {
        "value": false,
        "states": "186",
        "trans": "423",
        "depth": null
      },
      "Prop01Type": {
        "value": true,
        "states": "186",
        "trans": "423",
        "depth": "23"
      }
    },
    "Network01Bag": {
      "Prop02Safe": {
        "value": false,
        "states": null,
        "trans": null,
        "depth": "11"
      },
      "Prop04MsgSound": {
        "value": true,
        "states": "186",
        "trans": "423",
        "depth": "23"
      },
      "Prop03Sound": {
        "value": false,
        "states": "186",
        "trans": "423",
        "depth": null
      },
      "Prop01Type": {
        "value": true,
        "states": "186",
        "trans": "423",
        "depth": "23"
      }
    },
    "Network06Fifo": {
      "Prop02Safe": {
        "value": false,
        "states": null,
        "trans": null,
        "depth": "11"
      },
      "Prop04MsgSound": {
        "value": true,
        "states": "186",
        "trans": "423",
        "depth": "23"
      },
      "Prop03Sound": {
        "value": false,
        "states": "186",
        "trans": "423",
        "depth": null
      },
      "Prop01Type": {
        "value": true,
        "states": "186",
        "trans": "423",
        "depth": "23"
      }
    }
  }
}
